components/feature/session/src/main/java/mozilla/components/feature/session/FullScreenFeature.kt

```diff
@@ -39,12 +63,22 @@ open class FullScreenFeature(
      * @return Returns true if the fullscreen mode was successfully exited; false if no effect was taken.
      */
     override fun onBackPressed(): Boolean {
-        activeSession?.let {
-            if (it.fullScreenMode) {
-                sessionUseCases.exitFullscreen(it)
+        observation?.let {
+            if (it.inFullScreen) {
+                sessionUseCases.exitFullscreen(it.tabId)
                 return true
             }
         }
         return false
     }
 }
+
+private data class Observation(
```

Comment:
Some simple KDoc would be nice to have here that mentions Observation is used to just hold the fields that this feature cares about